api/types/types.go

```diff
@@ -340,22 +343,38 @@ type SummaryNetworkSettings struct {
 	Networks map[string]*network.EndpointSettings
 }
 
-// NetworkSettingsBase holds basic information about networks
+// NetworkSettingsBase holds the networking state of a specific container when inspecting it.
+//
+// Deprecated: this struct mostly contains deprecated fields and will be removed in v26.0.
```

Comment:
Don't think we can _remove_ the fields if we still support the older API versions.

Reply:
OH, I guess that's the approach with the duplicated fields 🤔

Reply:
How to completely get rid of these fields is still an unanswered question for me right now. One option could be through some clever use of struct tags and a custom JSON encoder, maybe.

But at least, this change clearly separates what's deprecated and what is not, make maintainers' life easier I believe.